src/loki/status_action.cc

```diff
@@ -1,8 +1,49 @@
+#include <boost/property_tree/json_parser.hpp>
+#include <iostream>
+#include <sstream>
+
+#include "baldr/tilehierarchy.h"
+#include "config.h"
 #include "loki/worker.h"
+#include "proto/status.pb.h"
 
 namespace valhalla {
 namespace loki {
-void loki_worker_t::status(Api&) const {
+void loki_worker_t::status(Api& request) const {
+
+  auto* status = request.mutable_status();
+
+  // only return more info if explicitly asked for (can be very expensive)
+  // bail if we wont be getting extra info
+  if (!request.options().verbose() || !allow_verbose)
+    return;
+
+  // get _some_ tile
+  static baldr::graph_tile_ptr tile = nullptr;
+  if (!tile) {
+    for (const auto& tile_id : reader->GetTileSet()) {
+      tile = reader->GetGraphTile(tile_id);
+      if (tile->id().level() < baldr::TileHierarchy::GetTransitLevel().level &&
+          tile->header()->nodecount() > 0) {
+        break;
+      }
+    }
+
+    if (connectivity_map) {
+      status->set_bbox(connectivity_map->to_geojson(2));
+    }
+
+    status->set_has_tiles(!!tile);
```

Comment:
```suggestion
    status->set_has_tiles(tile);
```

Reply:
well, like I said, the compiler doesn't like that:
```
[build] /home/nilsnolde/dev/cpp/valhalla/src/loki/status_action.cc:35:25: error: cannot convert ‘valhalla::baldr::graph_tile_ptr’ {aka ‘boost::intrusive_ptr<const valhalla::baldr::GraphTile>’} to ‘bool’
[build]    35 |   status->set_has_tiles(tile);
[build]       |                         ^~~~
[build]       |                         |
[build]       |                         valhalla::baldr::graph_tile_ptr {aka boost::intrusive_ptr<const valhalla::baldr::GraphTile>}
```

`!!tile` did work, but no idea why.. any suggestion how to make both you and the compiler happy? :)

Reply:
`status->set_has_tiles(static_cast<bool>(tile));` seems to compile

Reply:
its boolean conversion operator must be marked as explicit

Reply:
yeah it is: https://www.boost.org/doc/libs/1_76_0/libs/smart_ptr/doc/html/smart_ptr.html#intrusive_ptr

`explicit operator bool () const noexcept;`